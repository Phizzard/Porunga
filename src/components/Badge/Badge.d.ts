import React from 'react'
import { Palette, Rounded } from '../../types'
interface BadgeProps {
  /** The text to display in the badge */
  text?: string
  /** The background color of the badge */
  variant?: Palette
  /** An optional icon to display */
  icon?: React.ReactNode
  /** How round the corners are of the box */
  rounded?: Rounded
}
declare const Badge: ({ text, variant, rounded }: BadgeProps) => JSX.Element
export default Badge
