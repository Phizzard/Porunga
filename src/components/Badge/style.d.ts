/// <reference types="react" />
import { SerializedStyles } from '@emotion/core'
declare type BadgeProps = {
  color?: string
  backgroundColor?: string
  rounded?: SerializedStyles
}
export declare const StyledBadge: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').HTMLAttributes<HTMLSpanElement>,
    HTMLSpanElement
  >,
  BadgeProps,
  object
>
export {}
