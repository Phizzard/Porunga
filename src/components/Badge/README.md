Badges

```jsx
<Badge text="Nice" />
<Badge text="Nice" variant="success" />
<Badge text="Nice" variant="warning" />
<Badge text="Nice" variant="accent-1" />
```
