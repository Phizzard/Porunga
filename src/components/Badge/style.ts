import styled from '@emotion/styled'
import { SerializedStyles } from '@emotion/core'

type BadgeProps = {
  color?: string
  backgroundColor?: string
  rounded?: SerializedStyles
}

export const StyledBadge = styled.span<BadgeProps>`
  padding: 0.4rem 1.8rem;
  color: ${p => p.color};
  font-size: 0.8rem;
  font-weight: 500;
  background-color: ${p => p.backgroundColor};
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  ${p => p.rounded}
`
