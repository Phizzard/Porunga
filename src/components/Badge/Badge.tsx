import React from 'react'
import { useTheme } from 'emotion-theming'

import { get } from '../../utils/get'
import { ThemeConfig, Palette } from '../../types'
import { StyledBadge } from './style'

const COLOR_BLACK = '#000000'
const COLOR_WHITE = '#FFFFFFF'

interface BadgeProps {
  /** The text to display in the badge */
  text?: string
  /** The background color of the badge */
  variant?: Palette
  /** An optional icon to display */
  icon?: React.ReactNode
}

/**
 * This function tries to determine what font color to use based on contrast
 *
 * @param bgColor the background color of the badge
 */
function chooseTextColor(bgColor = ''): string {
  if (!bgColor) {
    return COLOR_BLACK
  }
  return parseInt(bgColor.replace('#', ''), 16) > 0xffffff / 2
    ? COLOR_BLACK
    : COLOR_WHITE
}

const Badge = ({ text = '', variant = 'brand' }: BadgeProps) => {
  const theme: ThemeConfig = useTheme()
  const backgroundColor = get(theme, 'root', 'colors', variant)
  const color = chooseTextColor(backgroundColor)

  return (
    <StyledBadge backgroundColor={backgroundColor} color={color}>
      {text}
    </StyledBadge>
  )
}

export default Badge
