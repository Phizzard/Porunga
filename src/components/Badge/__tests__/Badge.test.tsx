import React from 'react'
import Badge from '../Badge'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const component = renderer.create(<Badge />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
