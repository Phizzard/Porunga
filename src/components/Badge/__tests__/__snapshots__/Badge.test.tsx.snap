// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="css-k8b19o"
  color="#000000"
>
  
</span>
`;
